sdk/core/src/services/github/client/mod.rs

```diff
@@ -0,0 +1,284 @@
+                None => return Err(
+                    Error::GitHubErrorResponse(
+                        "==> Last hash is missing:".to_string()
+                    )
+                )
+            },
+            Err(err) => {
+                if let HyperError::Remote(status, msg) = err {
+                    return Err(
+                        Error::LastCommitHashError(status, msg)
+                    )
+                }
+
+                Commit {
+                    sha: None,
+                }
+            },
+        };
+
+        match gh_commits_rsp.sha {
+            Some(val) => Ok(Some(val)),
+            None => Ok(None),
+        }
+    }
+
+    /// Conventional Constructor.
+    pub fn new(pat: String) -> Self {
+        let token = format!("Bearer {}", pat);
+        Client { token }
+    }
+
+    /// Function to get the number of repos per page
+    pub async fn get_pages(&self, org: &String) -> Result<Vec<u32>, Error> {
+
+        let num_repos = self.get_num_pub_repos(org.to_string()).await?.unwrap_or(0);
+
+        println!("Number of Repositories in {org}: {:#?}", num_repos);
+
+        let num_calls = ((num_repos/100) as i8) + 1;
+        let num_last_call = num_repos % 100;
+
+        let mut vector = vec![100; usize::try_from(num_calls).unwrap()];
+
+        // This is crazy that it works.
+        *vector.last_mut().unwrap() = num_last_call;
+
+        Ok(vector)
+    }
+
+    /// Function to get the data for a page of repos
+    pub async fn get_page_of_repos(
+        &self,
+        org: &String,
+        page: usize,
+        per_page: &u32,
+    ) -> Result<Vec<Repo>, Error> {
+
+        let github_org_url = format!("{GH_URL}/orgs/{org}/repos?type=sources&page={page}&per_page={per_page}");
+
+        println!("Calling({})", github_org_url);
+
+        let response: Result<Option<Vec<Repo>>, HyperError> = get(
+            github_org_url.as_str(),
+            ContentType::Json,
+            self.token.as_str(),
+            None::<String>,
+        ).await;
+
+        match response {
+            Ok(option) => match option {
+                Some(value) => Ok(value),
+                None => Err(
+                    Error::GitHubErrorResponse(
+                        "Get request from GitHub had an empty response".to_string()
+                    )
+                ),
+            },
+            Err(err) => Err(
+                Error::GitHubResponse(err)
+            ),
+        }
+    }
+
+    /// Generates a unique clone path for a repository.
+    pub fn clone_path(&self, url: &str, hash: &String) -> String {
+
+        let repo_name = url
+            .split('/')
+            .collect::<Vec<&str>>()
+            .pop()
+            .unwrap()
+            .replace(".git", "");
+
+        format!("/harbor-output/{}/{}", hash, repo_name)
+    }
+
+    /// Clones a git repository to the specified clone path.
+    pub fn clone_repo(&self, clone_path: &str, url: &str) -> Result<(), Error> {
```

Comment:
So, I changed clone repo to this: 

```
    /// Clones a git repository to the specified clone path.
    pub fn clone_repo(&self, url: &str, last_hash: &String) -> Result<(), Error> {
        let clone_path = self.clone_path(&document.id, last_hash).as_str();
        self.client.clone_repo(clone_path, url)
    }
```

I am still believing that the client should do the actual cloning since it requires a method call, but making it a service functions feels like the correct move, does this address your finding?

Reply:
We'll create a git solution in plaftorm in the next sprint. This is A OK for this PR.